src/Sdk/DTLogging/Logging/ValueEncoders.cs

```diff
@@ -1,73 +1,97 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.ComponentModel;
+using System.Linq;
 using System.Security;
 using System.Text;
-using System.Text.RegularExpressions;
 using Newtonsoft.Json;
 
 namespace GitHub.DistributedTask.Logging
 {
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public delegate String ValueEncoder(String value);
+    public delegate IEnumerable<string> ValueEncoder(string value);
```

Comment:
Allow a single ValueEncode to emit multiple variations.  (In some cases, this avoids having to recompute the same intermediate data over-and-over.)